package/network/utils/wireguard-tools/files/wireguard.rpcd

```diff
@@ -0,0 +1,160 @@
+#!/bin/sh
+
+. /lib/functions.sh
+. /lib/functions/network.sh
+. /usr/share/libubox/jshn.sh
+
+proto_wireguard_setup_peer() {
+	local peer_config="$1"
+
+	local disabled
+	local public_key
+	local preshared_key
+	local allowed_ips
+	local route_allowed_ips
+	local endpoint_host
+	local endpoint_port
+	local persistent_keepalive
+
+	config_get device "${peer_config}" "device" "wg0"
+
+	if [ "$device" != "$name" ]; then
+		return 0
+	fi
+
+	config_get_bool disabled "${peer_config}" "disabled" 0
+	config_get public_key "${peer_config}" "public_key"
+	config_get preshared_key "${peer_config}" "preshared_key"
+	config_get allowed_ips "${peer_config}" "allowed_ips"
+	config_get_bool route_allowed_ips "${peer_config}" "route_allowed_ips" 0
+	config_get endpoint_host "${peer_config}" "endpoint_host"
+	config_get endpoint_port "${peer_config}" "endpoint_port"
```

Comment:
I would prefer to keep them separate if subsequent processes want to avoid the eventuality of pulling apart an IPv6 and port (should it ever come to that). Does wg internally do an SRV lookup (which sometimes provides ports) or only A/AAAA?